extensions/interactions/RatioExpressionInput/directives/oppia-interactive-ratio-expression-input.component.ts

```diff
@@ -17,112 +17,133 @@
-      ctrl.getWarningText = function() {
-        return errorMessage;
-      };
+  ngOnInit(): void {
+    const {
+      placeholder,
+      numberOfTerms
+    } = this.interactionAttributesExtractorService.getValuesFromAttributes(
+      'RatioExpressionInput',
+      {
+        placeholderWithValue: this.placeholderWithValue,
+        numberOfTermsWithValue: this.numberOfTermsWithValue
+      }
+    ) as RatioExpressionInputCustomizationArgs;
+    this.expectedNumberOfTerms = numberOfTerms.value;
+    this.placeholder = placeholder.value.unicode;
+    if (this.savedSolution !== undefined) {
+      let savedSolution = this.savedSolution;
+      savedSolution = Ratio.fromList(
+        savedSolution as RatioInputAnswer).toAnswerString();
+      this.answer = savedSolution;
+    } else {
+      this.answer = '';
+    }
+    const submitAnswerFn = () => this.submitAnswer();
+    const isAnswerValid = () => this.isAnswerValid();
+    this.currentInteractionService.registerCurrentInteraction(
+      submitAnswerFn, isAnswerValid);
+
+    setTimeout(
+      () => {
+        let focusLabel: string = this.labelForFocusTarget;
+        this.focusManagerService.setFocusWithoutScroll(focusLabel);
+      }, 0);
+  }
 
-      ctrl.submitAnswer = function(answer) {
-        try {
-          var ratioExpression =
-            Ratio.fromRawInputString(answer);
-          if (
-            (
-              ratioExpression.getNumberOfTerms() !==
-              ctrl.expectedNumberOfTerms) && (ctrl.expectedNumberOfTerms !== 0)
-          ) {
-            throw new Error(
-              'The creator has specified the number of terms in' +
-              ' the answer to be ' + ctrl.expectedNumberOfTerms + '.');
-          }
-          errorMessage = '';
-          ctrl.RatioExpressionInputForm.answer.$setValidity(
-            FORM_ERROR_TYPE, true);
-          CurrentInteractionService.onSubmit(
-            ratioExpression.getComponents(),
-            RatioExpressionInputRulesService);
-        } catch (parsingError) {
-          errorMessage = parsingError.message;
-          ctrl.RatioExpressionInputForm.answer.$setValidity(
-            FORM_ERROR_TYPE, false);
-        }
-      };
+  submitAnswer(): void {
+    try {
+      const ratioExpression =
+        Ratio.fromRawInputString(this.answer);
+      if (
+        (
+          ratioExpression.getNumberOfTerms() !==
+          this.expectedNumberOfTerms) && (this.expectedNumberOfTerms !== 0)
+      ) {
+        throw new Error(
+          'The creator has specified the number of terms in' +
+          ' the answer to be ' + this.expectedNumberOfTerms + '.');
+      }
+      this.errorMessage = '';
+      this.isValid = true;
+      this.currentInteractionService.onSubmit(
+        ratioExpression.getComponents() as unknown as string,
+        // eslint-disable-next-line max-len
+        this.ratioExpressionInputRulesService as unknown as InteractionRulesService);
```

Comment:
Why `as unknown`?

Reply:
actually we want argument as string, but `ratioExpression.getComponents()` is of type of number[], so we want to convert it into unknown first, then into string.
That's the reason.

Reply:
Then we should do the conversion explicitely through JS and not just with types, this will only introduce problems.

Reply:
as per 19-01-2022, angular migration team meeting we decided to solve this issue later. As many files of interactions are in this order. So cover them all togeather next week.

Reply:
Yeah sure, add a TODO then. I think @mridul-netizen already has an issue for this.

Reply:
https://github.com/oppia/oppia/issues/13015

Reply:
Done